spring-boot/src/main/java/org/springframework/boot/web/support/ErrorPageFilter.java

```diff
@@ -223,12 +223,12 @@ private String getErrorPath(Class<?> type) {
 		if (this.subtypes.containsKey(type)) {
 			return this.exceptions.get(this.subtypes.get(type));
 		}
-		Class<?> subtype = type;
-		while (subtype != Object.class) {
-			subtype = subtype.getSuperclass();
-			if (this.exceptions.containsKey(subtype)) {
-				this.subtypes.put(subtype, type);
-				return this.exceptions.get(subtype);
+		Class<?> supertype = type;
+		while (supertype != Object.class) {
+			supertype = supertype.getSuperclass();
+			if (this.exceptions.containsKey(supertype)) {
+				this.subtypes.put(type, supertype);
```

Comment:
This class has a nice battery of tests, but what's more important I _haven't changed the contract_ (_behaviour visible from the outside is still the same_), but rather fixed an _internal_ optimization. I can't see how I may test this particular _internal_ implementation detail without exposing it, which I think would be a bad thing. If you could shed some light on the way you'd wish such test to be implemented, I'd be most grateful.